src/oneD/MultiNewton.cpp

```diff
@@ -293,8 +316,14 @@ int MultiNewton::solve(double* x0, double* x1, OneDim& r, MultiJac& jac, int log
     double rdt = r.rdt();
     int j0 = jac.nEvals();
     int nJacReeval = 0;
-
+    int maxLoops = 1000;
+    int nloops = 0;
     while (true) {
+        if(nloops++ > maxLoops) {
+            return -1;
+        }
+        nloops++;
```

Comment:
I was running with the two-point method and the loop count was in the 60,000 range with no sign of ever getting out. It might just be a one-time thing, but getting stuck kind of pointed to a possible failure mode that wasn't handled in that function.